addons/hr_skills/models/hr_individual_skill.py

```diff
@@ -0,0 +1,260 @@
+        return self.env['hr.skill.type'].search([], limit=1)
+
+    skill_id = fields.Many2one('hr.skill', compute='_compute_skill_id', store=True, domain="[('skill_type_id', '=', skill_type_id)]", readonly=False, required=True, ondelete='cascade')
+    skill_level_id = fields.Many2one('hr.skill.level', compute='_compute_skill_level_id', domain="[('skill_type_id', '=', skill_type_id)]", store=True, readonly=False, required=True, ondelete='cascade')
+    skill_type_id = fields.Many2one('hr.skill.type',
+                                    default=_default_skill_type_id,
+                                    required=True, ondelete='cascade')
+    level_progress = fields.Integer(related='skill_level_id.level_progress')
+    color = fields.Integer(related="skill_type_id.color")
+    valid_from = fields.Date(string="Validity Start", default=fields.Date.today())
+    valid_to = fields.Date(string="Validity Stop")
+    number_of_levels = fields.Integer(related="skill_type_id.number_of_levels")
+    number_of_certified_type = fields.Integer(compute="_compute_number_of_certified_type", export_string_translation=False)
+    is_certification = fields.Boolean(related="skill_type_id.is_certification", export_string_translation=False)  # if is_certification change the model will not trigger the constrains
+    display_warning_message = fields.Boolean()
+
+    @api.constrains(lambda self: ['valid_from', 'valid_to', 'skill_id', 'skill_type_id', 'skill_level_id', self._linked_field_name])
+    def _check_not_overlapping_regular_skill(self):
+        domain = Domain.FALSE
+
+        for individual_skill in self:
+            ind_domain = Domain.AND([
+                Domain(f"{self._linked_field_name}.id", "=", individual_skill[self._linked_field_name].id),
+                Domain("skill_id.id", "=", individual_skill.skill_id.id),
+                Domain("id", "!=", individual_skill.id),
+            ])
+
+            if individual_skill.is_certification:
+                ind_domain = Domain.AND([
+                    ind_domain,
+                    Domain("skill_level_id.id", "=", individual_skill.skill_level_id.id),
+                    Domain('valid_from', '=', individual_skill.valid_from),
+                    Domain('valid_to', '=', individual_skill.valid_to),
+                ])
+            else:
+                ind_domain = Domain.AND([
+                    ind_domain,
+                    Domain('valid_from', '<=', individual_skill.valid_to),
+                    Domain.OR([
+                        Domain('valid_to', '=', False),
+                        Domain('valid_to', '>=', individual_skill.valid_to),
+                    ]),
+                ])
+
+            domain = Domain.OR([domain, ind_domain])
+        if self.env[self._name].search_count(domain, limit=1):
+            raise ValidationError(self.env._("At least one of your new records overlaps an existing one"))
```

Comment:
Could you provide more information for the end user ? Which record ? Which problematic dates ? 

Also, record is not very use friendly. Is it possible to use another label ?